### PR TITLE
Queued process support

### PR DESCRIPTION
**Pull request by andrewLab-lokalise**

### CODE REVIEW

File: svc_file.go [hunk 1]
```diff
@@ -142,6 +137,8 @@ func (c *FileService) Upload(projectID string, file FileUpload) (r FileUploadRes
 		file.CustomTranslationStatusInsertedKeys = Bool(true)
 	}
 
+	file.Queue = true
+
 	resp, err := c.post(c.Ctx(), fmt.Sprintf("%s/%s/%s/%s", pathProjects, projectID, pathFiles, "upload"), &r, file)
```
Comment (yarlson) [inline svc_file.go:142]: Response code from the endpoint could be either 202 or 302 (302 is not implemented yet). Both have a `Location` header which point to an endpoint to get current upload status.

Add location and status (202 - processing, 302 - file already uploaded) to the response object

File: svc_queuedprocesses.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+type QueuedProcessType string
+
+const (
+	FileImport   QueuedProcessType = "file-import"
+	SketchImport QueuedProcessType = "sketch-import"
+)
+
+func (qpt *QueuedProcessType) UnmarshalJSON(b []byte) error {
+	// Define a secondary type to avoid ending up with a recursive call to json.Unmarshal
```
Comment (yarlson) [inline svc_queuedprocesses.go:29]: You should not validate a server response in this case, a nice hack tho

**Comment by yarlson** [comment]
Breaking changes, you should bump a major version in `go.mod`

`module github.com/lokalise/go-lokalise-api/v3`

File: svc_queuedprocesses.go (hunk 2 above)
Comment (andrewLab-lokalise) [reply svc_queuedprocesses.go]: Updated

File: svc_file.go (hunk 1 above)
Comment (andrewLab-lokalise) [reply svc_file.go:142]: As agreed, statuses and Location header will not be part of logic. Using response body to get process ID for final result obtain in future

**Pull request merged**